### DISCUSSION TITLE
How to query data by date/time (ISO8601 format) range

### DISCUSSION THREAD

**Discussion by rnarayan** [supabase/supabase#2351] [Questions] [answered]
I am building a calendar functionality in a NextJs project. Calendar will have events. How do I query events for a particular date or a date range (week or month).

**Fields in Events Table:**
 ```
   Title: text,
   startTime: timestamptz,
   endTime: timestamptz
```

Time is stored is ISO8601 format. for example, _"2021-07-13T22:00:41.445Z"_

I would like to query data like this: 
```
let { data, error } = await supabase
  .from('events')
  .select("*")
  .eq('startTime', '2021-07-13')
```

Or 

```
let { data, error } = await supabase
  .from('events')
  .select("*")
  .gte('startTime', '2021-07-01')
  .lt('startTime', '2021-08-01')
```

The query above doesn't work. I don't have any SQL experience but learning while working with Supabase. Any help will be appreciated. Thanks.

**Comment by steve-chavez** [comment, marked as answer]
Hey @rnarayan,

Can you try with:

```js
let { data, error } = await supabase
  .from('events')
  .select("*")
  .eq('startTime', new Date('2021-07-13').toISOString())
```

**Reply by rnarayan** [reply]
A little variation of the code above worked. I had to use gte and lte.

```
const { data, error} = await supabase
      .from('events')
      .select('*')
      .gte('starttime', new Date('2021-07-10').toISOString())
      .lte('starttime', new Date('2021-07-11').toISOString())
```

The code above gave me all the events for the date 2021-07-10.  
 eq filter didn't return any data because it tries to match the time too, which should be the expected behavior.

Thanks a lot for help. It worked.
